src/adapter/repository/gitRemoteService.ts

```diff
@@ -0,0 +1,88 @@
+import { Repository } from './git.d';
+import { Branch } from '../../types';
+import { IGitCommandExecutor } from '..';
+import { GitOriginType } from '.';
+
+export class GitRemoteService {
+    constructor(private readonly repo: Repository, private readonly gitCmdExecutor: IGitCommandExecutor) {}
+    private get currentBranch(): string {
+        return this.repo.state.HEAD!.name || '';
+    }
+    public async getBranchesWithRemotes(): Promise<Branch[]> {
```

Comment:
Use `git remote show <name> -n` to get a list of branches associated with a remote for `pull`